### PR TITLE
Misc small chains improvements

### PR DESCRIPTION
**Pull request by simonbaird**
A handful of miscellaneous fixes and cleanups related to chains. It's mostly tweaks to the demo scripts, but there's one configuration change intended to allow argocd admin permissions in the tekton-chains namespace.

See commits for explanations.

### CODE REVIEW

**Comment by simonbaird** [comment]
FYI @Michkov @sbose78 re the top commit with "allow-argocd-to-manage" .

File: hack/chains/README.md
Comment (simonbaird) [inline hack/chains/README.md:62]: I'd like to explain what is needed for cluster-bot, but I'm not sure the answer...

File: hack/chains/default-pipelines-demo.sh
Comment (simonbaird) [inline hack/chains/default-pipelines-demo.sh:19]: FYI the awkward syntax is so I can leave on the set -u

File: hack/chains/kaniko-cosign-verify.sh
Comment (simonbaird) [inline hack/chains/kaniko-cosign-verify.sh:15]: It's defined already in _helpers.sh

File: hack/chains/rekor-image-lookup.sh
Comment (simonbaird) [inline hack/chains/rekor-image-lookup.sh:20]: This lookup method was first described in [HACBS-75](https://issues.redhat.com/browse/HACBS-75) - thanks @caugello .

File: hack/chains/rekor-image-lookup.sh
Comment (simonbaird) [inline hack/chains/rekor-image-lookup.sh:10]: This lookup method was first described in [HACBS-75](https://issues.redhat.com/browse/HACBS-75) - thanks @caugello .

**Comment by simonbaird** [comment]
FYI @lcarva, @joejstuart, @bcaton85, @zregvart, @robnester-rh, @caugello .

File: hack/chains/rekor-verify-taskrun.sh
Comment (lcarva) [inline hack/chains/rekor-verify-taskrun.sh:41]: Should we use `values[0]` here and in the line below since this script assumes a single value?

File: hack/chains/rekor-verify-taskrun.sh
Comment (lcarva) [inline hack/chains/rekor-verify-taskrun.sh:53]: fyi `jq` itself can decode base64: `echo $ATTESTATION_DATA | jq -r '.data | @base64d | @base64d' | yq-pretty`

File: hack/chains/default-pipelines-demo.sh
Comment (lcarva) [inline hack/chains/default-pipelines-demo.sh:21]: Maybe `--ignore-not-found=true` might be useful?

**Review by lcarva — APPROVED** [review]
Looks great! I just left some minor suggestions which are not to be considered a blocker for this PR.

File: components/build/tekton-chains/allow-argocd-to-manage.yaml
Comment (simonbaird) [reply components/build/tekton-chains/allow-argocd-to-manage.yaml:13]: I moved it to #193 so we can discuss/review separately from the rest of these tweaks.

File: hack/chains/rekor-image-lookup.sh
Comment (simonbaird) [reply hack/chains/rekor-image-lookup.sh:7]: Ack, thanks. I'll go for the --no-tags option, but I learned something from the second options, thanks!

File: hack/chains/rekor-verify-taskrun.sh
Comment (simonbaird) [reply hack/chains/rekor-verify-taskrun.sh:41]: It's probably safe since I think there's only one key/value, but I think you're right for the sake of readability and explicitness. I'll change it.

File: hack/chains/rekor-verify-taskrun.sh
Comment (simonbaird) [reply hack/chains/rekor-verify-taskrun.sh:53]: That's neat! I didn't know that. I don't want to change it here, but maybe later we could use that trick in a few places to save some processes.

File: hack/chains/default-pipelines-demo.sh
Comment (simonbaird) [reply hack/chains/default-pipelines-demo.sh:21]: Thanks!

**Comment by simonbaird** [comment]
New revision:
* Apply sugestions from @lcarva 
* Add some more to the readme, particularly related to the DEPLOY_ONLY
* Remove the argocd patch, it's now in #193 
* Improved commit messages in a couple of places
* Split bash helper into its own commit

**Comment by simonbaird** [comment]
I might merge this soon if there are no objections.

File: hack/chains/rekor-verify-taskrun.sh
Comment (simonbaird) [reply hack/chains/rekor-verify-taskrun.sh:43]: Hmm, actually putting [0] there doesn't work... Will figure it out later..

**Comment by simonbaird** [comment]
I ran all the chains demos in my local CRC and it looks good to me. Got a +1 in slack from @lcarva . This is just chains demo tweaks, so we can merge without review from outside contract team. I'm pretty happy that we're now looking at two transparency log entries for a build-push task. Let's merge.

**Pull request merged**
